File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/objects/ObjectGroundAddedMessage.scala





















// Generated on 05/03/2013 10:34:45
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ObjectGroundAddedMessage extends MessageDeserializer {
  val messageId = 3017

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ObjectGroundAddedMessage.messageId

    val cellId: Short
    val objectGID: Short

    def serialize(buf: Buffer) {
      buf.writeShort(cellId)
      buf.writeShort(objectGID)
    }

  }

  protected def create(cellId: Short, objectGID: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val cellId = buf.readShort()
    val objectGID = buf.readShort()

    create(cellId, objectGID)
  }

}

object ObjectGroundAddedMessage extends ObjectGroundAddedMessage {
  case class MessageImpl(cellId: Short, objectGID: Short) extends Message
  type Target = MessageImpl

  override def create(cellId: Short, objectGID: Short) = MessageImpl(cellId, objectGID)

  def apply(cellId: Short, objectGID: Short) = create(cellId, objectGID)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ObjectGroundAddedMessage.java


// Generated on 05/08/2013 19:37:47
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ObjectGroundAddedMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 3017;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short cellId;
    public short objectGID;
    
    public ObjectGroundAddedMessage() { }
    
    public ObjectGroundAddedMessage(short cellId, short objectGID) {
        this.cellId = cellId;
        this.objectGID = objectGID;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(cellId);
        buf.writeShort(objectGID);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        cellId = buf.readShort();
        if (cellId < 0 || cellId > 559)
            throw new RuntimeException("Forbidden value on cellId = " + cellId + ", it doesn't respect the following condition : cellId < 0 || cellId > 559");
        objectGID = buf.readShort();
        if (objectGID < 0)
            throw new RuntimeException("Forbidden value on objectGID = " + objectGID + ", it doesn't respect the following condition : objectGID < 0");
    }
    
}
